packages/fxa-content-server/app/tests/spec/views/mixins/cached-credentials-mixin.js

```diff
@@ -62,22 +62,20 @@ describe('views/mixins/cached-credentials-mixin', () => {
       assert.isTrue(view.isPasswordNeededForAccount(account));
     });
 
-    it('asks for password if the relier wants keys (Sync)', () => {
-      account.set('email', 'testuser@testuser.com');
-      sinon.stub(relier, 'wantsKeys').callsFake(() => true);
+    it('asks for a password if the account has no sessionToken', () => {
+      account.unset('sessionToken');
+      sinon.stub(relier, 'wantsKeys').callsFake(() => false);
       sinon.stub(user, 'isSyncAccount').callsFake(() => true);
       model.unset('chooserAskForPassword');
-      sinon
-        .stub(view, 'getPrefillEmail')
-        .callsFake(() => 'testuser@testuser.com');
+      sinon.stub(view, 'getPrefillEmail').callsFake(() => '');
 
       assert.isTrue(view.isPasswordNeededForAccount(account));
     });
 
```

Comment:
I don't think there's a test to verify that `it('does not ask for a password if the account has email and sessionToken')`, so I'd suggest adding one.

Reply:
Nice catch and it pointed out that the tests weren't always testing what we expected!

Reply:
i.e., tests weren't failing when I expected them to fail.